src/pages/cw-storage-plus/basics.mdx

```diff
@@ -0,0 +1,82 @@
+import { Callout } from "nextra/components";
+
+# Basics
+
+## Containers
+
+_cw-storage-plus_ provides a number of storage containers that can be used to
+store data on the blockchain. The fundamental ones are
+
+- [`Item`](containers/item),
+- [`Map`](containers/map),
+- [`Deque`](containers/deque).
+
+An [`Item`](containers/item) is a simple container that stores a single value.
+The other two are a little more involved - they are collections capable of
+storing multiple values, and provide several methods to interact with them.
+
+## Keys and prefixes
+
+One task of a storage library like `cw-storage-plus` is to manage the namespace
+of keys provided by the blockchain.
+
+When constructing a container, you must provide a key of type `&'static str`.
+This usually means you'll be providing a string literal or some constant.
+
+In the case of an [`Item`](containers/item), the provided key is the exact key
+(a standard UTF-8 string) under which the value will be saved. In the case of
+collections, the provided key is really a prefix; the collection will use this
```

Comment:
maybe say that this is a "namespace"?

Reply:
I would find "the provided key is really a namespace" a bit confusing here if that's what you mean. Are you worried the name of the parameter (`namespace`) in constructors will be confusing?

Reply:
I'm not sure what exactly is a bit confusing - but vaguely that in collections, keys are not typically what you think of as the namespace/prefix, but rather the thing you pass to save/load/etc.

how about: "Collections are created with a namespace which acts as a prefix for all keys; the collection will use this..." ?

Reply:
The problem I'm having is that AFAIK a namespace is a set of names, while (for us) a prefix is a thing that generates that set of names.

Would this work?

```md
In the case of an [`Item`](containers/item), the provided string is the exact key
(a standard UTF-8 string) under which the value will be saved. In the case of
collections, the provided string is a prefix (called `namespace` in code); the collection will use this
```